install.sh

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env bash
+set -e
+
+version="2.2.0"
+
+# Do nothing if echidna is already installed
+if echidna-test --version > /dev/null 2>&1
+then
+  echo "WARNING: $(echidna-test --version) is installed but is out-of-date, we recommend removing it"
+fi
+
+# Do nothing if echidna is already installed
+if echidna --version > /dev/null 2>&1
+then
+  echo "$(echidna --version) is already installed" && exit
+  # TODO: check that the version is up to date, otherwise offer to upgrade
+fi
+
+# In the unlikely scenario that python3 isn't available, let the user install it manually
+if ! python3 --version > /dev/null 2>&1
+then echo "python3 is not available, please install it manually before attempting to install echidna" && exit 1
+fi
+
+# install pip3 if not already installed
+if ! pip3 --version > /dev/null 2>&1
+then
+  echo "pip3 is not available, installing it now.."
+  curl https://bootstrap.pypa.io/get-pip.py | python3
+fi
+
+# install slither if not already installed
+if ! slither --version > /dev/null 2>&1
+then
+  echo "slither is not available, installing it now.."
+  pip3 install slither-analyzer --user
```

Comment:
It would be cool to have slither updated if it's out of date. PATH may also need to be adjusted to account for slither installed with `--user`.

Maybe we should use a (fresh?) venv instead and just link `slither{,-*}` and `crytic-compile` to the directory we then add to the PATH for echidna?